src/main.rs

```diff
@@ -334,27 +333,107 @@ struct InputAction {
 
     /// Did the user ask to run tests or benchmarks?
     build_kind: BuildKind,
+
+    // Name of the built binary
+    bin_name: String,
 }
 
 impl InputAction {
     fn manifest_path(&self) -> PathBuf {
         self.pkg_path.join("Cargo.toml")
     }
 
+    fn script_path(&self) -> PathBuf {
+        self.pkg_path.join("main.rs")
+    }
+
     fn cargo(
         &self,
-        build_kind: BuildKind,
+        action: &InputAction,
         script_args: &[String],
         run_quietly: bool,
     ) -> MainResult<Command> {
-        cargo(
-            build_kind,
-            &self.manifest_path().to_string_lossy(),
-            self.toolchain_version.as_deref(),
-            &self.metadata,
-            script_args,
-            run_quietly,
-        )
+        let release_mode = !self.metadata.debug && !matches!(action.build_kind, BuildKind::Bench);
+
+        let built_binary_path = platform::binary_cache_path()
+            .join(if release_mode { "release" } else { "debug" })
+            .join(&action.bin_name);
```

Comment:
Since this lacks `.exe`, won't this always be considered unbuilt for Windows?

Reply:
@epage It does indeed! Fixed in https://github.com/fornwall/rust-script/pull/94 and release in version 0.24.0, thanks a lot!